### PR TITLE
Introduce AUR, Logging, Loading, and more

### PR DESCRIPTION
**Pull request by lucemans**

### CODE REVIEW

File: .deploy/bump.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (m1guelpf) [inline .deploy/bump.sh:1]: Can we move this to .github, to clean up the file structure a lil?

File: Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [package]
 license = "MIT"
 edition = "2021"
-version = "0.1.0"
+version = "0.1.4-pre8"
```
Comment (m1guelpf) [inline Cargo.toml:4]: pre8?

File: src/main.rs [hunk 3]
```diff
@@ -36,7 +39,16 @@ async fn main() -> Result<(), ()> {
         .filter_level(cli.verbose.log_level_filter())
         .init();
 
-    let api_token = std::env::var("OPENAI_API_KEY").expect("No API key specified.");
+    println!(
+        "Auto-commit by Miguel Piedrafita v{}\n---",
+        env!("CARGO_PKG_VERSION")
+    );
+
```
Comment (m1guelpf) [inline src/main.rs]: lets get rid of this I think

File: src/main.rs [hunk 4]
```diff
@@ -57,11 +71,46 @@ async fn main() -> Result<(), ()> {
...
+        Spinners::Point,
+        Spinners::Pong,
+        Spinners::Runner,
+        Spinners::SoccerHeader,
+        Spinners::Speaker,
+        Spinners::SquareCorners,
+        Spinners::Triangle,
+    ];
+
+    let spinner = vs.choose(&mut rand::thread_rng()).unwrap().clone();
```
Comment (m1guelpf) [inline src/main.rs:96]: tf is happening here lol

File: Cargo.toml (hunk 2 above)
Comment (lucemans) [reply Cargo.toml:4]: It auto bumps :shrug:

File: src/main.rs (hunk 4 above)
Comment (lucemans) [reply src/main.rs:96]: Hehehehe, beautifull stuff

File: Cargo.toml (hunk 2 above)
Comment (m1guelpf) [reply Cargo.toml:4]: yea but why not .5 why pre-8

File: src/main.rs [hunk 5]
```diff
@@ -44,6 +44,11 @@ async fn main() -> Result<(), ()> {
         std::process::exit(1);
     });
 
+    // Using git2 check if the repository exists
+    let repository = git2::Repository::open_from_env().map_err(|_| {
+            eprintln!("It looks like you are not in a git repository.\nPlease run this command from the root of a git repository, or initialize one using `git init`.");
+            std::process::exit(1);
+        }).unwrap();
```
Comment (m1guelpf) [inline src/main.rs:51]: If we're gonna pull in git2, maybe we can use for committing as well? Not blocking now but may be interesting later

**Pull request merged**